City-of-Helsinki/helsinki-paatokset
# PP-291: Search tuning part 1

Added more fields to indices
Added custom proccessors for fields
Fix frontpage search form to pass params correctly
Bumped paatokset_search application to version 0.2.2

To test:

Make sure you have decision / issue content imported before testing.

- Remove old files to (just to make sure composer actually installs the new ones) `rm -rf vendor/paatokset`. Also remove old asset files to make sure they're deleted `rm -rf public/modules/custom/paatokset_search/assets/*`.
- Run `make shell` and inside container `composer update paatokset/paatokset_search --prefer-source`
- Either run `make new` or `drush cr; drush cim -y`. 
- Check your [Search API decisions-index's settings](https://helsinki-paatokset.docker.so/fi/admin/config/search/search-api/index/decisions). You should see 2 new fields there, `decisions_url` and `color_class`.
- Check the [index status](https://helsinki-paatokset.docker.so/fi/admin/config/search/search-api/index/decisions). All content should be queued for reindexing. If not, press the `Queue all items for reindexing` -button and then run run the indexing proccess by clicking `Index now`-button, (Edit: sometimes you need to rebuild the tracking information. This is the case if you have decision content imported, but the index status indicates 0/0 content is indexed)
- Go to your frontpage. If you don't have search form hero added, add it from edit page -> add hero -> select paatokset search form as paragraph type. After you have the search form hero added, write a search phrase and submit. The search page should render with the inputted search phrase filled in.
- Remove the query from the search page url  and refresh. The search form should now render with no text input and the search results should be hidden.
- Make a new search. Make sure that the search results have the correct color as the highight color on top of the result card and you can navigate to decision / issue pages by clicking the cards.
